### PR TITLE
Restructure SDE kernel

### PR DESCRIPTION
**Pull request by frankschae**
- simplifies the SDE kernel.
- adds fully allocating regression function.
- makes a closer connection to Mitosis for the tilde kernel to support time dependence of its parameters in a follow up.
- wraps tests in better readable test sets

### CODE REVIEW

File: src/regression.jl [hunk 1]
```diff
@@ -1,35 +1,76 @@
...
+    t = t2
+    y = y2
+    Γ = Γ + zi*ds
+
+  end
+  return Mitosis.Gaussian{(:F,:Γ)}(μ, Γ)
+end
 
-        ds = t2 - t
+function conjugate(r::Regression!, Y, prior::Gaussian)
```
Comment (frankschae) [inline src/regression.jl:42]: @mschauer  This conjugate function takes at the moment out-of-place functions `g` from `SDEProblem`  but does all the operations for the regression method inplace. Should this be changed to take only inplace functions `f!` or should this somehow support both?
Out of place functions  `g` could still be used with  `conjugate(r::Regression,..)` which however then also allocates.

Comment (mschauer) [reply src/regression.jl:42]: Can't immediately tell, can we make our minds up later or is there lock in?

File: src/types.jl [hunk 2]
```diff
@@ -1,32 +1,56 @@
...
 
-function SDEKernel(f,g,tstart,tend,pest,plin;dt=nothing)
-  SDEKernel{typeof(f),typeof(g),typeof(tstart),
-            typeof(dt),typeof(pest),typeof(plin)}(f,g,tstart,tend,dt,pest,plin)
+struct Message{kernelType,solType,sol2Type,tType}
+  ktilde::kernelType
+  sol::solType
+  soldis::sol2Type
+  ts::tType
 end
```
Comment (mschauer) [inline src/types.jl:13]: Nice. Quoting our paper:  while other choices for the message are sometimes convenient, they are essentially equivalent: the format of the message `m` is something private between the forward map and the backward map. Having a type here makes sure that both sides agree on the interpretation

**Review by mschauer — APPROVED** [review]

File: src/regression.jl (hunk 1 above)
Comment (frankschae) [reply src/regression.jl:42]: no big urgency at the moment. There is anyway a random test failure with the regression that I have to solve first before we can merge this PR.  I was just looking into the #10 and saw that SDEProblems with inplace functions are still missing and I wasn't sure which option one should prefer.

**Pull request merged**